user-guide/Troubleshooting/Procedures/Troubleshooting_STaaS_Issues.md

```diff
@@ -0,0 +1,110 @@
+```
+
+To resolve this issue, go to Task Manager and restart the *DataMiner CloudGateway* service.
+
+## The session token has expired
+
+Under normal circumstances, CloudGateway refreshes the cloud session automatically. However, if **CloudGateway is down for longer than three days**, for example because the server is down, the cloud session will become invalid. This will cause DataMiner startup to fail for systems using STaaS.
+
+When you encounter this issue, you will find entries similar to the examples below in the *SLCloudStorage.txt* log file:
+
+```txt
+CloudSettings could not be retrieved from the cloud. Retrying in 00:00:05. Exception: SLCloudStorage.Repositories.Exceptions.CloudSettingsRepositoryException: Failed to do GetCloudAccessTokenRequest. Received the following error messages: { "message": "The Service Principal of this DMS is expired (3/14/2023 8:09:51 AM +00:00) but should soon be refreshed automatically." }
+```
+
+```txt
+CloudSettings could not be retrieved from the cloud. Retrying in 00:00:05. Exception: SLCloudStorage.Repositories.Exceptions.CloudSettingsRepositoryException: Exception while doing GetCcaGatewayConfigRequest. ---> System.AggregateException: One or more errors occurred. ---> DataMinerMessageBroker.API.Exceptions.SubscriptionException: No responders are available for the request. ---> NATS.Client.NATSNoRespondersException: No responders are available for the request.
+```
+
+To resolve this issue:
+
+1. [Open SLNetClientTest tool](xref:Opening_the_SLNetClientTest_tool) on the DMA.
+
+   > [!WARNING]
+   > Always be extremely careful when using this tool, as it can have far-reaching consequences on the functionality of your DataMiner System.
+
+1. Select *Offline Tools* > *CcaGateway (offline)* > *Renew cloud session* and complete the renew process.
+
+   > [!NOTE]
+   > As the renewal of the Service Principal (SP) token is managed by a cloud service, it can take a few minutes before the renewal is fully synced.
+
+1. Wait a few minutes and then restart the DMA.
+
+   The issue should now be resolved.
+
+> [!NOTE]
+> If the DataMiner System consists of multiple DMAs, it is sufficient to do this on one of the DMAs.
+
+## STaaS is unreachable
+
+If STaaS is unreachable for some reason, the following error message will be shown in the Alarm Console:
+
+![Error in Alarm Console](~/user-guide/images/STaaS_troubleshooting_Alarm_Console.png)
+
+DataMiner will automatically initiate offload mode, ensuring that no data loss occurs. While this error is displayed in the Alarm Console, **do not restart** the system or any of its elements.
+
+This issue will typically resolves itself. If it persists, contact support.
```

Comment:
Which address should they contact? staas@dataminer.services?

Reply:
yes 👍